content/blog/pulumi-loves-python/index.md

```diff
@@ -0,0 +1,115 @@
+---
+title: "Pulumi Loves Python"
+date: 2024-07-02T16:17:38+02:00
+draft: false
+meta_desc: First-class Python support with Poetry, Pythonic input types and built-in type checking.
+meta_image: meta.png
+authors:
+    - julien-poissonnier
+tags:
+    - python
+social:
+    twitter:
+    linkedin:
+
+---
+
+Pulumi has first-class support for Python, and it is one of our most popular languages. We are always working to improve the Python experience, and we are excited to announce some new features that make it even easier to use Python with Pulumi, including support for Poetry, built-in type checking, and more Pythonic input types.
+
+<!--more-->
+
+<!-- expand on the introduction here -->
+
+## Poetry
+
+Pulumi has long had support to create and manage virtual environments using Python's builtin package manager [pip](https://pip.pypa.io/en/stable/). With the [latest release of Pulumi](https://github.com/pulumi/pulumi/releases/tag/v3.121.0), we are excited to announce that we now support [Poetry](https://python-poetry.org) as well. Poetry is a popular Python dependency management tool that allows you to declare your dependencies in a simple and concise way and manage your virtual environment with ease.
+
+When creating a new Pulumi project, the Pulumi CLI will now ask you if you want to use Poetry to manage your dependencies. If you choose to use Poetry, the Pulumi CLI will automatically create a new Poetry project for you and install the necessary dependencies. To opt-in to using Poetry for an existing Pulumi project, set the [`toolchain` runtime option](https://www.pulumi.com/docs/concepts/projects/project-file/#runtime-options) to `poetry` and run `pulumi install`.
+
+```yaml
+name: python-and-poetry-are-best-friends
+runtime:
+  name: python
+  options:
+    toolchain: poetry
+```
+
+## Built-in type checking
+
+Pulumi Python SDKs include type hints compatible with type checkers such as [MyPy](https://www.mypy-lang.org) and [Pyright](https://microsoft.github.io/pyright/#/). In the latest release of Pulumi, you can ask Pulumi to run your typechecker of choice for you as part of Pulumi operations and fail if there are any type errors. This can help you catch type errors earlier and ensure that your Pulumi programs are type-safe.
+
+Add the typechecker of your choice to [your dependencies](https://www.pulumi.com/docs/languages-sdks/python/#packages) and set the [`typechecker` runtime option](https://www.pulumi.com/docs/concepts/projects/project-file/#runtime-options) in your project file to `mypy` or `pyright` to enable automatic type checking.
+
+```yaml
+name: python-with-typechecking
+runtime:
+  name: python
+  options:
+    typechecker: pyright
```

Comment:
I suggest we add a link somewhere in this blog to a full working example of a Pulumi app configured with poetry+pyright to show all the needed configuration in one place.

Reply:
I created a PR to update the aws-py-fargate example to use all the latest features https://github.com/pulumi/examples/pull/1656